packrat

```diff
@@ -96,22 +99,27 @@ function create_dir() {
 function run_collectors() {
     local DEST="${1}"
 
-    collect_distro_info "${DEST}" > "${DEST}/distro.log" 2>&1 &
-    collect_package_manager_info "${DEST}" > "${DEST}/package-manager.log" 2>&1 &
-    collect_env_info "${DEST}" > "${DEST}/env.log" 2>&1 &
-    collect_cpu_info "${DEST}" > "${DEST}/cpu.log" 2>&1 &
-    collect_device_info "${DEST}" > "${DEST}/device.log" 2>&1 &
-    collect_system_info "${DEST}" > "${DEST}/system.log" 2>&1 &
-    collect_kernel_info "${DEST}" > "${DEST}/kernel.log" 2>&1 &
-    collect_block_info "${DEST}" > "${DEST}/block.log" 2>&1 &
-    collect_memory_info "${DEST}" > "${DEST}/memory.log" 2>&1 &
-    collect_net_info "${DEST}" > "${DEST}/net.log" 2>&1 &
-    collect_cgroup_info "${DEST}" > "${DEST}/cgroup.log" 2>&1 &
-    collect_libvirt_info "${DEST}" > "${DEST}/libvirt.log" 2>&1 &
-    collect_firewall_info "${DEST}" > "${DEST}/firewall.log" 2>&1 &
-    collect_container_info "${DEST}" > "${DEST}/container.log" 2>&1 &
-
-    wait
+    collectors=("distro" "package_manager" "env" "cpu" "device" "system" "kernel" "block" "memory" "net" "cgroup" "libvirt" "firewall" "container")
+    collector_pids=()
+
+    echo "Total collectors: ${#collectors[@]}"
+    for collector in "${collectors[@]}"; do
+	echo "Launching '${collector}' collector"
+	collect_${collector}_info "${DEST}" > "${DEST}/${collector}.log" 2>&1 &
+	collector_pid=${!}
+	collector_pids+=("${collector_pid}")
+    done
+
+    total_rc=0
+    for idx in $( seq 0 $(( ${#collector_pids[@]} - 1)) ); do
+	echo -n "Waiting for '${collectors[${idx}]}' collector with PID ${collector_pids[${idx}]} to exit..."
+	wait ${collector_pids[${idx}]}
+	collector_rc=${?}
+	echo "exited with return code ${collector_rc}"
+	(( total_rc += ${collector_rc} ))
+    done
+
+    return ${total_rc}
```

Comment:
looks good! I assume now we can toggle on/off something like STOP_ON_PACKRAT_FAIL

Reply:
I haven't made it optional so far, but I suppose we could if the need arises.